lib/token-translator.js

```diff
@@ -123,8 +123,9 @@ TokenTranslator.prototype = {
                  type === tt.arrow || type === tt.jsxTagStart ||
                  type === tt.incDec || type === tt.starstar ||
                  type === tt.jsxTagEnd || (type.binop && !type.keyword) ||
-                 type.isAssign) {
-
+                 type.isAssign ||
+                 (type.prefix && (token.value === "~" || token.value === "!"))
```

Comment:
I think you can safely say if the first character of the token is not an identifier start, then it's a punctuator.